pyproject.toml

```diff
@@ -77,35 +77,35 @@ dependencies = [
 
 [project.optional-dependencies]
 test = [
-    "pytest == 7.1.2",
-    "pytest-xdist == 2.1.0",
-    "pytest-timeout == 1.4.2",
-    "pytest-cov == 2.10.1",
-    "nbval == 0.9.3",
-    "IPython >= 8.10.0, <8.12.1",
-    "PyYAML == 6.0.1",
+    "pytest >= 7.1.2",
+    "pytest-xdist >= 2.1.0",
+    "pytest-timeout >= 1.4.2",
+    "pytest-cov >= 2.10.1",
+    "nbval >= 0.9.3",
+    "IPython >= 8.10.0",
+    "PyYAML >= 6.0.1",
     "coverage[toml] >= 6.4",
 ]
 dev = [
-    "ruff == 0.0.228",
-    "darglint == 1.8.0",
+    "ruff >= 0.0.228",
+    "darglint >= 1.8.0",
     "pre-commit >= 2.20.0",
     "evalml[docs,test]",
 ]
 updater = [
     "alteryx-open-src-update-checker >= 2.1.0"
 ]
 docs = [
-    "docutils >=0.15.2, < 0.17",
+    "docutils >= 0.15.2, < 0.17",
     "pydata-sphinx-theme >= 0.3.1",
     "astroid <= 2.6.6",
     "Sphinx >= 5.0.0",
     "nbconvert >= 6.5.0",
     "nbsphinx >= 0.8.5, < 0.9.0",
```

Comment:
A combination of nbsphinx, astroid, and docutils cause the `readthedocs` pipeline to fail. I'll need to open a separate ticket to uncap those.